Fix for long corrpower analytic

spficklin opened the pull request:
This is a fix for the corrpower analytic.  As described in issue #126, this analytic would take an extreme amount of time to complete.   The problem was that it was not using a correct index for each working block and would repeatedly process the same elements.  To fix this, I added a `seek` function to the `pairwise_matrix_pair.cpp` file that is used to properly find the index for each block.

spficklin committed:
Fix for issue #126

bentsherman commented:
@spficklin @4ctrl-alt-del I am concerned that this solution is a bit more complicated than it needs to be. Now that we have a couple of analytics that use MPI (conditional-test, corrpower, similarity), I'm going to review them to see if there is a best practice that will apply to all of them. Then I will get back to you about this PR.

4ctrl-alt-del committed:
Fixed correlation power bug.

After much investigation discovered a bug in the work block of the
correlation power analytic that would cause it to fail in MPI mode because
the read interface would not read in the proper start index, causing every
work block to start at the default pair index of (1,0).

spficklin commented:
@bentsherman the cond-test doesn't break under MPI but it's not working properly.  It has a similar problem that needs the fix in this PR.  See issue #133.  The similarity analytic is fine because it works through every cell of the matrix.  The cond-test and the corrpower have to deal with a sparse matrix.

spficklin commented:
Oh, I should note, The commit that @4ctrl-alt-del just made is to correct issue #134. Thanks @4ctrl-alt-del !

bentsherman commented:
I just pushed some changes that should fix the issues you've been having with corrpower and cond-test, so this PR is no longer needed.
